TombEngine/Objects/TR3/Entity/tr3_civvy.cpp

```diff
@@ -79,341 +119,484 @@ namespace TEN::Entities::Creatures::TR3
-				if (AI.ahead)
-					extraHeadRot.y = AI.angle;
+					creature.MaxTurn = 0;
+					
+					if (AI.ahead)
+						jointHeadRot.y = targetAngle;
+					else if (item.AIBits & GUARD)
+						jointHeadRot.y = AIGuard(&creature);
+
+					if (creature.Alerted || Random::TestProbability(CIVVY_WAIT_CHANCE))
+						item.Animation.TargetState = CIVVY_STATE_IDLE;
 
-				if (item->AIBits & GUARD)
-					item->Animation.TargetState = CIVVY_WAIT;
-				else if (creature->Mood == MoodType::Escape)
-				{
-					if (Lara.TargetEntity != item && AI.ahead)
-						item->Animation.TargetState = CIVVY_STATE_IDLE;
 					break;
-				}
-				else if ((item->AIBits & FOLLOW) && (creature->ReachedGoal || laraAI.distance > SQUARE(SECTOR(2))))
-					item->Animation.TargetState = CIVVY_STATE_IDLE;
-				else if (creature->Mood == MoodType::Bored)
-					item->Animation.TargetState = CIVVY_STATE_WALK_FORWARD;
-				else if (AI.ahead && AI.distance < CIVVY_WALK_RANGE)
-					item->Animation.TargetState = CIVVY_STATE_WALK_FORWARD;
 
-				break;
+				case CIVVY_STATE_RUN:
 
-			case CIVVY_AIM0:
-				creature->MaxTurn = CIVVY_WALK_TURN_RATE_MAX;
-				creature->Flags = 0;
+					creature.MaxTurn = CIVVY_RUN_TURN_RATE_MAX;
+					tilt = angle / 2;
 
-				if (AI.ahead)
-				{
-					extraTorsoRot.x = AI.xAngle;
-					extraTorsoRot.y = AI.angle;
-				}
+					if (AI.ahead)
+						jointHeadRot.y = AI.angle;
 
-				if (AI.bite && AI.distance < CIVVY_ATTACK0_RANGE)
-					item->Animation.TargetState = CIVVY_PUNCH0;
-				else
-					item->Animation.TargetState = CIVVY_STATE_IDLE;
+					if (item.AIBits & GUARD)
+						item.Animation.TargetState = CIVVY_STATE_IDLE;
+					else if (creature.Mood == MoodType::Escape)
+						item.Animation.TargetState = CIVVY_STATE_RUN;
+					//else if (creature.Mood == MoodType::Escape && Lara.TargetEntity != &item && AI.ahead)
+						//item.Animation.TargetState = CIVVY_STATE_IDLE;
+					else if ((item.AIBits & FOLLOW) && (creature.ReachedGoal || targetDistance > SQUARE(SECTOR(2))))
+						item.Animation.TargetState = CIVVY_STATE_IDLE;
+					else if (creature.Mood == MoodType::Bored)
+						item.Animation.TargetState = CIVVY_STATE_WALK;
+					else if (AI.ahead && AI.distance <= CIVVY_WALK_RANGE)
+						item.Animation.TargetState = CIVVY_STATE_WALK;
 
-				break;
+					break;
 
-			case CIVVY_AIM1:
-				creature->MaxTurn = CIVVY_WALK_TURN_RATE_MAX;
-				creature->Flags = 0;
+				case CIVVY_STATE_AIM_CLOSE_PUNCH:
 
-				if (AI.ahead)
-				{
-					extraTorsoRot.x = AI.xAngle;
-					extraTorsoRot.y = AI.angle;
-				}
+					creature.MaxTurn = CIVVY_AIM_TURN_RATE_MAX;
+					creature.Flags = 0;
 
-				if (AI.ahead && AI.distance < CIVVY_ATTACK1_RANGE)
-					item->Animation.TargetState = CIVVY_PUNCH1;
-				else
-					item->Animation.TargetState = CIVVY_STATE_IDLE;
+					if (AI.ahead)
+					{
+						jointTorsoRot.x = AI.xAngle;
+						jointTorsoRot.y = AI.angle;
+					}
 
-				break;
+					if ((creature.Enemy->IsLara() && AI.bite && AI.distance < CIVVY_ATTACK_CLOSE_PUNCH_RANGE) ||
```

Comment:
Same here, don't use 2 value, adjust the original one to work for both.

Reply:
Done